docs/administration-guide/databases/bigquery.md

```diff
@@ -1,25 +1,111 @@
+
+## Google Cloud Platform: creating a service account and JSON file
+
+You'll first need a [service account](https://cloud.google.com/iam/docs/service-accounts) JSON file that Metabase can use to access your BigQuery dataset. Service accounts are intended for non-human users (such as applications like Metabase) to authenticate (who am I?) and authorize (what can I do?) their API calls.
+
+To create the service account JSON file, follow Google's documentation on [setting up a service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts) for your BigQuery dataset. Here's the basic flow:
+
+1. **Create service account**. From your Google Cloud Platform project console, open the main sidebar menu on the left, go to the **IAM & Admin** section, and select **Service account**. The console will list existing service accounts, if any. At the top of the screen, click on **+ CREATE SERVICE ACCOUNT**.
+
+2. **Fill out the service account details**. Name the service account, and add a description (the service account ID will populate once you add a name.) Then click the **Create** button.
+
+3. **Grant service account access to this project**. You'll need to add **roles** to the service account so that Metabase will have permission to view and run queries against your dataset. Make sure you add the following roles to the service account:
+
+    - BigQuery Data Viewer
+    - BigQuery Metadata Viewer
+    - BigQuery Job User (distinct from BigQuery User)
+
+    For more information on **roles** in BigQuery, see [Google Cloud Platorm's documentation](https://cloud.google.com/bigquery/docs/access-control).
+
+4. **Create key**. Once you have assigned roles to the service account, click on the **Create Key** button, and select **JSON** for the **key type**. The JSON file will download to your computer.
+
+> **You can only download the key once**. If you delete the key, you'll need to create another service account with the same roles.
+
+## Metabase: adding a BigQuery dataset
+
+Once you have your created and downloaded your service account JSON file for your BigQuery dataset, head over to your Metabase instance, click on the **settings** cog, and select **Admin** to bring up Admin mode. In the **Databases** section, click on the **Add database** button in the upper right. (You can also access this page from the **Settings** menu, under **Setup** and **GET CONNECTED**.)
+
+On the **ADD DATABASE** page, select **BigQuery** from the **Database type** dropdown. Metabase will present you with the relevant configuration fields to fill out:
+
+![images](../images/bigquery_add_database.png)
+
+### Fields
```

Comment:
This separation of "fields" vs "sliders" seems odd and unnecessary to me. I think you can just list all the settings in order without breaking them out like this.

Reply:
Agreed. I'll put everything under an umbrella **Settings** section.